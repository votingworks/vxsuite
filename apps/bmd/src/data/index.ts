import { join } from 'path'
import { loadElectionDefinition } from '@votingworks/fixtures'

export const electionSampleDefinition = loadElectionDefinition(
  join(__dirname, './electionSample.json')
)
export const electionSampleNoSealDefinition = loadElectionDefinition(
  join(__dirname, './electionSampleNoSeal.json')
)
export const electionSampleWithSealDefinition = loadElectionDefinition(
  join(__dirname, './electionSampleWithSeal.json')
)
export const electionPrimarySampleDefinition = loadElectionDefinition(
  join(__dirname, './electionPrimarySample.json')
)
