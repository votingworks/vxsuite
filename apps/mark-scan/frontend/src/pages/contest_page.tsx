import { CandidateVote } from '@votingworks/types';
import { Screen, Prose, P, Font, LinkButton } from '@votingworks/ui';
import { singlePrecinctSelectionFor } from '@votingworks/utils';
import React, { useContext } from 'react';
import { assert, throwIllegalValue } from '@votingworks/basics';
import { useHistory, useParams } from 'react-router-dom';
import { Contest as MarkFlowContest } from '@votingworks/mark-flow-ui';
import styled from 'styled-components';
import { BallotContext } from '../contexts/ballot_context';
import { ElectionInfo } from '../components/election_info';
import { Sidebar } from '../components/sidebar';
import { screenOrientation } from '../lib/screen_orientation';
import { ButtonFooter } from '../components/button_footer';
import { DisplaySettingsButton } from '../components/display_settings_button';

interface ContestParams {
  contestNumber: string;
}

const Breadcrumbs = styled.div`
  padding: 0 0.5rem;
`;

export function ContestPage(): JSX.Element {
  const { contestNumber } = useParams<ContestParams>();
  const history = useHistory();
  const isReviewMode = history.location.hash === '#review';

  const {
    ballotStyleId,
    contests,
    electionDefinition,
    machineConfig,
    precinctId,
    updateVote,
    votes,
  } = useContext(BallotContext);

  // eslint-disable-next-line vx/gts-safe-number-parse
  const currentContestIndex = parseInt(contestNumber, 10);
  const contest = contests[currentContestIndex];

  const prevContestIndex = currentContestIndex - 1;
  const prevContest = contests[prevContestIndex];

  const nextContestIndex = currentContestIndex + 1;
  const nextContest = contests[nextContestIndex];

  assert(
    electionDefinition,
    'electionDefinition is required to render ContestPage'
  );
  assert(
    typeof precinctId === 'string',
    'precinctId is required to render ContestPage'
  );
  const { isLandscape, isPortrait } = screenOrientation(machineConfig);

  const vote = votes[contest.id];

  const ballotContestNumber = currentContestIndex + 1;
  const ballotContestsLength = contests.length;

  const isVoteComplete = (() => {
    switch (contest.type) {
      case 'yesno':
        return !!vote;
      case 'candidate':
        return contest.seats === ((vote as CandidateVote) ?? []).length;
      case 'ms-either-neither':
        return (
          votes[contest.pickOneContestId]?.length === 1 ||
          votes[contest.eitherNeitherContestId]?.[0] === 'no'
        );
      /* istanbul ignore next */
      default:
        throwIllegalValue(contest);
    }
  })();

  const nextContestButton = (
    <LinkButton
      id="next"
      variant={isVoteComplete ? 'next' : 'nextSecondary'}
      aria-label="next contest"
      to={nextContest ? `/contests/${nextContestIndex}` : '/review'}
    >
      Next
    </LinkButton>
  );

  const previousContestButton = (
    <LinkButton
      variant="previous"
      large={isPortrait}
      id="previous"
      aria-label="previous contest"
      to={prevContest ? `/contests/${prevContestIndex}` : '/'}
    >
      Back
    </LinkButton>
  );

  const reviewScreenButton = (
    <LinkButton
      large
      variant={isVoteComplete ? 'next' : 'nextSecondary'}
      to={`/review#contest-${contest.id}`}
      id="next"
    >
      Review
    </LinkButton>
  );

  const settingsButton = <DisplaySettingsButton />;

  return (
    <Screen navRight={isLandscape}>
      {isPortrait && (
        <Breadcrumbs>
          <P align="right">
            Contest <Font weight="bold">{ballotContestNumber}</Font> of{' '}
            <Font weight="bold">{ballotContestsLength}</Font>
          </P>
        </Breadcrumbs>
      )}
      <MarkFlowContest
        election={electionDefinition.election}
        contest={contest}
        votes={votes}
        updateVote={updateVote}
      />
      {isPortrait ? (
        <ButtonFooter>
          {isReviewMode ? (
            <React.Fragment>
              {settingsButton}
              {reviewScreenButton}
            </React.Fragment>
          ) : (
            <React.Fragment>
              {previousContestButton}
              {settingsButton}
              {nextContestButton}
            </React.Fragment>
          )}
        </ButtonFooter>
      ) : (
        <Sidebar
          footer={
            <React.Fragment>
              <ButtonFooter>{settingsButton}</ButtonFooter>
              <ElectionInfo
                electionDefinition={electionDefinition}
                ballotStyleId={ballotStyleId}
                precinctSelection={singlePrecinctSelectionFor(precinctId)}
                horizontal
              />
            </React.Fragment>
          }
        >
          <Prose>
            <P align="center">
              Contest <Font weight="bold">{ballotContestNumber}</Font> of{' '}
              <Font weight="bold">{ballotContestsLength}</Font>
            </P>
            {isReviewMode ? (
              <P>{reviewScreenButton}</P>
            ) : (
              <React.Fragment>
                <P>{nextContestButton}</P>
                <P>{previousContestButton}</P>
              </React.Fragment>
            )}
          </Prose>
        </Sidebar>
      )}
    </Screen>
  );
}
