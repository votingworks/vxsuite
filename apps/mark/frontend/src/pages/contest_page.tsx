import { CandidateVote } from '@votingworks/types';
import { Screen, LinkButton, useScreenInfo } from '@votingworks/ui';
import React, { useContext } from 'react';
import { assert, throwIllegalValue } from '@votingworks/basics';
import { useHistory, useParams } from 'react-router-dom';
import { Contest as MarkFlowContest } from '@votingworks/mark-flow-ui';
import { BallotContext } from '../contexts/ballot_context';
import { ButtonFooter } from '../components/button_footer';
import { DisplaySettingsButton } from '../components/display_settings_button';

interface ContestParams {
  contestNumber: string;
}

export function ContestPage(): JSX.Element {
  const { contestNumber } = useParams<ContestParams>();
  const history = useHistory();
  const isReviewMode = history.location.hash === '#review';

  const { contests, electionDefinition, precinctId, updateVote, votes } =
    useContext(BallotContext);

  const screenInfo = useScreenInfo();

  // eslint-disable-next-line vx/gts-safe-number-parse
  const currentContestIndex = parseInt(contestNumber, 10);
  const contest = contests[currentContestIndex];

  const prevContestIndex = currentContestIndex - 1;
  const prevContest = contests[prevContestIndex];

  const nextContestIndex = currentContestIndex + 1;
  const nextContest = contests[nextContestIndex];

  assert(
    electionDefinition,
    'electionDefinition is required to render ContestPage'
  );
  assert(
    typeof precinctId === 'string',
    'precinctId is required to render ContestPage'
  );

  const vote = votes[contest.id];

  const ballotContestNumber = currentContestIndex + 1;
  const ballotContestsLength = contests.length;

  const isVoteComplete = (() => {
    switch (contest.type) {
      case 'yesno':
        return !!vote;
      case 'candidate':
        return contest.seats === ((vote as CandidateVote) ?? []).length;
      case 'ms-either-neither':
        return (
          votes[contest.pickOneContestId]?.length === 1 ||
          votes[contest.eitherNeitherContestId]?.[0] === 'no'
        );
      /* istanbul ignore next */
      default:
        throwIllegalValue(contest);
    }
  })();

  const nextContestButton = (
    <LinkButton
      id="next"
      variant={isVoteComplete ? 'next' : 'nextSecondary'}
      aria-label="next contest"
      to={nextContest ? `/contests/${nextContestIndex}` : '/review'}
    >
      Next
    </LinkButton>
  );

  const previousContestButton = (
    <LinkButton
      variant="previous"
      id="previous"
      aria-label="previous contest"
      to={prevContest ? `/contests/${prevContestIndex}` : '/'}
    >
      Back
    </LinkButton>
  );

  const reviewScreenButton = (
    <LinkButton
      large
      variant={isVoteComplete ? 'next' : 'nextSecondary'}
      to={`/review#contest-${contest.id}`}
      id="next"
    >
      Review
    </LinkButton>
  );

  const settingsButton = <DisplaySettingsButton />;

  return (
    <Screen navRight={!screenInfo.isPortrait}>
      <MarkFlowContest
        breadcrumbs={{
          ballotContestCount: ballotContestsLength,
          contestNumber: ballotContestNumber,
        }}
        election={electionDefinition.election}
        contest={contest}
        votes={votes}
        updateVote={updateVote}
      />
      <ButtonFooter>
        {isReviewMode ? (
          <React.Fragment>
            {settingsButton}
            {reviewScreenButton}
          </React.Fragment>
        ) : (
          <React.Fragment>
            {previousContestButton}
            {settingsButton}
            {nextContestButton}
          </React.Fragment>
        )}
      </ButtonFooter>
    </Screen>
  );
}
