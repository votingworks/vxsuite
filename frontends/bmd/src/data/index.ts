import { safeParseElectionDefinition } from '@votingworks/types';
import * as electionPrimarySample from './electionPrimarySample.json';
import * as electionSample from './electionSample.json';
import * as electionSampleNoSeal from './electionSampleNoSeal.json';
import * as electionSampleWithSeal from './electionSampleWithSeal.json';
import * as electionSampleWithSealAndReportingUrl from './electionSampleWithSealAndReportingUrl.json';

/**
 * Election definition from `./electionPrimarySample.json`.
 */
export const electionPrimarySampleDefinition = safeParseElectionDefinition(
  electionPrimarySample.asText()
).unsafeUnwrap();

/**
 * Election definition from `./electionSample.json`.
 */
export const electionSampleDefinition = safeParseElectionDefinition(
  electionSample.asText()
).unsafeUnwrap();

/**
 * Election definition from `./electionSampleNoSeal.json`.
 */
export const electionSampleNoSealDefinition = safeParseElectionDefinition(
  electionSampleNoSeal.asText()
).unsafeUnwrap();

/**
 * Election definition from `./electionSampleWithSeal.json`.
 */
export const electionSampleWithSealDefinition = safeParseElectionDefinition(
  electionSampleWithSeal.asText()
).unsafeUnwrap();

/**
 * Election definition from `./electionSampleWithSealAndReportingUrl.json`.
 */
export const electionSampleWithSealAndReportingUrlDefinition =
  safeParseElectionDefinition(
    electionSampleWithSealAndReportingUrl.asText()
  ).unsafeUnwrap();
