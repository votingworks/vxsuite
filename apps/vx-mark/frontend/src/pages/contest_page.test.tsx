import React from 'react';
import { screen } from '@testing-library/react';
import { Route } from 'react-router-dom';
import { fakeMachineConfig } from '../../test/helpers/fake_machine_config';

import { render as renderWithBallotContext } from '../../test/test_utils';
import { electionSampleDefinition } from '../data';

import { ContestPage } from './contest_page';

const electionSample = electionSampleDefinition.election;
const firstContestTitle = electionSample.contests[0].title;

it('Renders ContestPage', () => {
  const { container } = renderWithBallotContext(
    <Route path="/contests/:contestNumber" component={ContestPage} />,
    {
      route: '/contests/0',
      precinctId: electionSample.precincts[0].id,
      ballotStyleId: electionSample.ballotStyles[0].id,
    }
  );
  screen.getByText(firstContestTitle);
  expect(container).toMatchSnapshot();
});

it('Renders ContestPage in Landscape orientation', () => {
  const { container } = renderWithBallotContext(
    <Route path="/contests/:contestNumber" component={ContestPage} />,
    {
      route: '/contests/0',
      precinctId: electionSample.precincts[0].id,
      ballotStyleId: electionSample.ballotStyles[0].id,
      machineConfig: fakeMachineConfig({ screenOrientation: 'landscape' }),
    }
  );
  screen.getByText(firstContestTitle);
  expect(container).toMatchSnapshot();
});

it('Renders ContestPage in Landscape orientation in Review Mode', () => {
  window.location.hash = '#review';
  renderWithBallotContext(
    <Route path="/contests/:contestNumber" component={ContestPage} />,
    {
      route: '/contests/0',
      precinctId: electionSample.precincts[0].id,
      ballotStyleId: electionSample.ballotStyles[0].id,
      machineConfig: fakeMachineConfig({ screenOrientation: 'landscape' }),
    }
  );
  screen.getByText(firstContestTitle);
  screen.getByText('Review');
});
