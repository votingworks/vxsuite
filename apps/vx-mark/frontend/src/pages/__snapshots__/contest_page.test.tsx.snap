// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ContestPage 1`] = `
.c7 {
  position: relative;
  border-radius: 0.125rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  background: #FFFFFF;
  cursor: pointer;
  padding: 0.5rem 0.5rem 0.5rem 4rem;
  text-align: left;
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

.c7::before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  border-right: 1px solid;
  border-color: rgb(211,211,211);
  border-radius: 0.125rem 0 0 0.125rem;
  background: #ffffff;
  width: 3rem;
  text-align: center;
  color: #028099;
  font-size: 2rem;
  font-weight: 700;
  content: '';
}

.c1 {
  margin: 0 auto;
  width: 100%;
  padding: 1rem 30px 0.5rem;
}

.c2 {
  text-transform: uppercase;
  font-size: 0.85rem;
  font-weight: 600;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  overflow: auto;
}

.c3::before,
.c3::after {
  position: absolute;
  z-index: 1;
  width: 100%;
  height: 0.25rem;
  content: '';
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c3::before {
  top: 0;
  opacity: 0;
  background: linear-gradient( to bottom, rgb(177,186,190) 0%, transparent 100% );
}

.c3::after {
  bottom: 0;
  opacity: 0;
  background: linear-gradient( to bottom, transparent 0%, rgb(177,186,190) 100% );
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c5 {
  margin: 0 auto;
  width: 100%;
  padding: 0.25rem 20px 0;
}

.c6 {
  display: grid;
  grid-auto-rows: minmax(auto,1fr);
  grid-gap: 20px;
  margin-bottom: 40px;
}

.c9 {
  white-space: nowrap;
}

.c9::after {
  display: inline-block;
  margin-left: 0.5rem;
  background: url('/images/arrow-right-open-black.svg') no-repeat;
  width: 1.375rem;
  height: 1.375rem;
  vertical-align: text-bottom;
  content: '';
}

.c10 {
  white-space: nowrap;
}

.c10::before {
  display: inline-block;
  margin-right: 0.5rem;
  background: url('/images/arrow-left-open-black.svg') no-repeat;
  width: 1.375rem;
  height: 1.375rem;
  vertical-align: text-bottom;
  content: '';
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #333333;
  padding: 20px;
  color: #ffffff;
  gap: 20px;
}

.c8 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c8 > *:first-child {
  -webkit-flex: 2 1;
  -ms-flex: 2 1;
  flex: 2 1;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c8 button {
  padding-right: 10px;
  padding-left: 10px;
}

.c0 {
  padding: 30px 30px 0;
}

@media (min-width:480px) {
  .c7 {
    padding: 1rem 1rem 1rem 4rem;
  }
}

<div>
  <div
    class="sc-hUpaWb kraGuc"
  >
    <div
      class="c0"
    >
      <div
        class="sc-hKwCoD gmiWky"
      >
        <p
          class="sc-bdvvaa iybVtA"
        >
          This is the
           
          <strong>
            1st
             contest
          </strong>
           of
           
          22 contests
           on your ballot.
        </p>
      </div>
    </div>
    <main
      class="sc-kLwgWK kRCmMs"
    >
      <div
        class="c1"
        id="contest-header"
      >
        <div
          class="sc-hKwCoD gmiWky"
          id="audiofocus"
        >
          <h1
            aria-label="United States President and Vice-President."
          >
            <div
              class="c2"
            >
              United States
            </div>
            President and Vice-President
          </h1>
          <p>
            <span
              class="sc-bdvvaa NLRzK"
            >
              Vote for 
              1
              .
            </span>
             
            <span
              class="screen-reader-only"
            >
              To navigate through the contest choices, use the down button. To move to the next contest, use the right button.
            </span>
          </p>
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <button
                aria-label=" Joseph Barchi and Joseph Hallaren, Federalist."
                class="c7"
                data-choice="barchi-hallaren"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Joseph Barchi and Joseph Hallaren
                    </strong>
                    <br />
                    Federalist
                  </p>
                </div>
              </button>
              <button
                aria-label=" Adam Cramer and Greg Vuocolo, People's."
                class="c7"
                data-choice="cramer-vuocolo"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Adam Cramer and Greg Vuocolo
                    </strong>
                    <br />
                    People's
                  </p>
                </div>
              </button>
              <button
                aria-label=" Daniel Court and Amy Blumhardt, Liberty."
                class="c7"
                data-choice="court-blumhardt"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Daniel Court and Amy Blumhardt
                    </strong>
                    <br />
                    Liberty
                  </p>
                </div>
              </button>
              <button
                aria-label=" Alvin Boone and James Lian, Constitution."
                class="c7"
                data-choice="boone-lian"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Alvin Boone and James Lian
                    </strong>
                    <br />
                    Constitution
                  </p>
                </div>
              </button>
              <button
                aria-label=" Ashley Hildebrand-McDougall and James Garritty, Whig."
                class="c7"
                data-choice="hildebrand-garritty"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Ashley Hildebrand-McDougall and James Garritty
                    </strong>
                    <br />
                    Whig
                  </p>
                </div>
              </button>
              <button
                aria-label=" Martin Patterson and Clay Lariviere, Labor."
                class="c7"
                data-choice="patterson-lariviere"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Martin Patterson and Clay Lariviere
                    </strong>
                    <br />
                    Labor
                  </p>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
    <nav
      class="c8"
    >
      <button
        aria-label="next contest"
        class="sc-egiSv iFSayO"
        id="next"
        role="option"
        type="button"
      >
        <span
          class="c9"
        >
          Next
        </span>
      </button>
      <button
        aria-label="previous contest"
        class="sc-egiSv iFSayO"
        id="previous"
        role="option"
        type="button"
      >
        <span
          class="c10"
        >
          Back
        </span>
      </button>
      <button
        aria-label="Change Settings"
        class="sc-egiSv iFSayO"
        type="button"
      >
        Settings
      </button>
    </nav>
  </div>
</div>
`;

exports[`Renders ContestPage in Landscape orientation 1`] = `
.c6 {
  position: relative;
  border-radius: 0.125rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  background: #FFFFFF;
  cursor: pointer;
  padding: 0.5rem 0.5rem 0.5rem 4rem;
  text-align: left;
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

.c6::before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  border-right: 1px solid;
  border-color: rgb(211,211,211);
  border-radius: 0.125rem 0 0 0.125rem;
  background: #ffffff;
  width: 3rem;
  text-align: center;
  color: #028099;
  font-size: 2rem;
  font-weight: 700;
  content: '';
}

.c0 {
  margin: 0 auto;
  width: 100%;
  padding: 1rem 30px 0.5rem;
}

.c1 {
  text-transform: uppercase;
  font-size: 0.85rem;
  font-weight: 600;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  overflow: auto;
}

.c2::before,
.c2::after {
  position: absolute;
  z-index: 1;
  width: 100%;
  height: 0.25rem;
  content: '';
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c2::before {
  top: 0;
  opacity: 0;
  background: linear-gradient( to bottom, rgb(177,186,190) 0%, transparent 100% );
}

.c2::after {
  bottom: 0;
  opacity: 0;
  background: linear-gradient( to bottom, transparent 0%, rgb(177,186,190) 100% );
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c4 {
  margin: 0 auto;
  width: 100%;
  padding: 0.25rem 20px 0;
}

.c5 {
  display: grid;
  grid-auto-rows: minmax(auto,1fr);
  grid-gap: 20px;
  margin-bottom: 40px;
}

.c15 {
  max-width: 250px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1.5rem 1rem 0;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
}

.c14 div:first-child {
  margin-right: 1rem;
  min-width: 5rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1;
  -ms-flex: 0 1;
  flex: 0 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #333333;
  color: #ffffff;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
}

.c8 p > button {
  width: 100%;
}

.c11 {
  margin: 1rem;
}

.c9 {
  white-space: nowrap;
}

.c9::after {
  display: inline-block;
  margin-left: 0.5rem;
  background: url('/images/arrow-right-open-black.svg') no-repeat;
  width: 1.375rem;
  height: 1.375rem;
  vertical-align: text-bottom;
  content: '';
}

.c10 {
  white-space: nowrap;
}

.c10::before {
  display: inline-block;
  margin-right: 0.5rem;
  background: url('/images/arrow-left-open-black.svg') no-repeat;
  width: 1rem;
  height: 1rem;
  vertical-align: text-bottom;
  content: '';
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 20px;
}

.c12 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c12 button {
  padding-right: 10px;
  padding-left: 10px;
}

@media (min-width:480px) {
  .c6 {
    padding: 1rem 1rem 1rem 4rem;
  }
}

<div>
  <div
    class="sc-hUpaWb fvcRs"
  >
    <main
      class="sc-kLwgWK kRCmMs"
    >
      <div
        class="c0"
        id="contest-header"
      >
        <div
          class="sc-hKwCoD gmiWky"
          id="audiofocus"
        >
          <h1
            aria-label="United States President and Vice-President."
          >
            <div
              class="c1"
            >
              United States
            </div>
            President and Vice-President
          </h1>
          <p>
            <span
              class="sc-bdvvaa NLRzK"
            >
              Vote for 
              1
              .
            </span>
             
            <span
              class="screen-reader-only"
            >
              To navigate through the contest choices, use the down button. To move to the next contest, use the right button.
            </span>
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <button
                aria-label=" Joseph Barchi and Joseph Hallaren, Federalist."
                class="c6"
                data-choice="barchi-hallaren"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Joseph Barchi and Joseph Hallaren
                    </strong>
                    <br />
                    Federalist
                  </p>
                </div>
              </button>
              <button
                aria-label=" Adam Cramer and Greg Vuocolo, People's."
                class="c6"
                data-choice="cramer-vuocolo"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Adam Cramer and Greg Vuocolo
                    </strong>
                    <br />
                    People's
                  </p>
                </div>
              </button>
              <button
                aria-label=" Daniel Court and Amy Blumhardt, Liberty."
                class="c6"
                data-choice="court-blumhardt"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Daniel Court and Amy Blumhardt
                    </strong>
                    <br />
                    Liberty
                  </p>
                </div>
              </button>
              <button
                aria-label=" Alvin Boone and James Lian, Constitution."
                class="c6"
                data-choice="boone-lian"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Alvin Boone and James Lian
                    </strong>
                    <br />
                    Constitution
                  </p>
                </div>
              </button>
              <button
                aria-label=" Ashley Hildebrand-McDougall and James Garritty, Whig."
                class="c6"
                data-choice="hildebrand-garritty"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Ashley Hildebrand-McDougall and James Garritty
                    </strong>
                    <br />
                    Whig
                  </p>
                </div>
              </button>
              <button
                aria-label=" Martin Patterson and Clay Lariviere, Labor."
                class="c6"
                data-choice="patterson-lariviere"
                data-selected="false"
                role="option"
                type="button"
              >
                <div
                  class="sc-hKwCoD gmiWky"
                >
                  <p
                    class="sc-bdvvaa eSpfPW"
                  >
                    <strong>
                      Martin Patterson and Clay Lariviere
                    </strong>
                    <br />
                    Labor
                  </p>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
    <nav
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="sc-hKwCoD gmiWky"
        >
          <p
            class="sc-bdvvaa cFJatN"
          >
            This is the 
            <strong>
              1st
            </strong>
             of
             
            22 contests
            .
          </p>
          <p>
            <button
              aria-label="next contest"
              class="sc-egiSv iFSayO"
              id="next"
              role="option"
              type="button"
            >
              <span
                class="c9"
              >
                Next
              </span>
            </button>
          </p>
          <p>
            <button
              aria-label="previous contest"
              class="sc-egiSv feoAXt"
              id="previous"
              role="option"
              type="button"
            >
              <span
                class="c10"
              >
                Back
              </span>
            </button>
          </p>
        </div>
      </div>
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          <button
            aria-label="Change Settings"
            class="sc-egiSv dHUDgX"
            type="button"
          >
            Settings
          </button>
        </div>
        <div
          aria-hidden="true"
          class="c13"
          data-testid="election-info"
        >
          <div
            class="c14"
          >
            <div
              aria-hidden="true"
              class="c15"
            />
            <div
              class="sc-hKwCoD eDCLhO"
            >
              <h5
                aria-label=" General Election."
              >
                 General Election
              </h5>
              <p
                class="sc-bdvvaa iybVtA"
              >
                November 3, 2020
                <br />
                <span
                  class="sc-gsDJrp fUIxJs"
                >
                  Franklin County
                  ,
                </span>
                 
                <span
                  class="sc-gsDJrp fUIxJs"
                >
                  State of Hamilton
                </span>
              </p>
              <p
                class="sc-bdvvaa kKHwYn"
              >
                <span
                  class="sc-gsDJrp fUIxJs"
                >
                  Center Springfield
                  , 
                </span>
                <span
                  class="sc-gsDJrp fUIxJs"
                >
                  ballot style 
                  12
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
